crates/executor/src/blockchain_tree/block_indices.rs

```diff
@@ -0,0 +1,258 @@
+//! Implementation of [`BlockIndices`] related to [`super::BlockchainTree`]
+
+use super::chain::{Chain, ChainId, ForkBlock};
+use reth_primitives::{BlockHash, BlockNumber, SealedBlockWithSenders};
+use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
+
+/// Internal indices of the block.
+#[derive(Clone)]
+pub struct BlockIndices {
```

Comment:
this is basically the block graph that has all the indices etc?

Reply:
Something like that, but it is specialized to tree structure.